web/usuaris/src/context/AuthProvider.tsx

```diff
@@ -0,0 +1,61 @@
+import React, {
+  createContext,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react'
+
+import { urls } from '../constants'
+
+export type TAuthContext = {
+  isAuth: boolean
+  children: React.ReactNode
+  setIsAuth: (isAuth: boolean) => void
+  error: string
+  setError: (error: string) => void
+}
+
+const authContext = createContext<TAuthContext | null>(null)
+
+export const useAuth = () => {
+  const context = useContext(authContext)
+  if (!context) {
+    throw new Error('useAuth must be used within an AuthProvider')
+  }
+
+  return context
+}
+
+export const AuthProvider: React.FC<
+  Omit<TAuthContext, 'isAuth' | 'setIsAuth' | 'error' | 'setError'>
+> = ({ children }) => {
+  const [isAuth, setIsAuth] = useState<boolean>(false)
+  const [error, setError] = useState('')
+
+  const value = useMemo(
+    () => ({ children, isAuth, setIsAuth, error, setError }),
+    [children, isAuth, error]
+  )
+
+  useEffect(() => {
+    fetch(urls.logIn)
```

Comment:
Is the same endpoint for login? 
This is a POST endpoint that needs dni & pwd in the request body, and you're trying to request it with a GET, so... 
![2024-04-04 (1)](https://github.com/IT-Academy-BCN/ita-wiki/assets/92027716/6bdb2bb8-cfc4-417e-9ac3-47ed8f635ab7)

Reply:
True, I didn't check that... I will have a look.